lib/core/base/virtual-node.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Wrap the real node and provide list of the flattened children
+ *
+ * @param node {Node} - the node in question
+ * @param shadowId {String} - the ID of the shadow DOM to which this node belongs
+ */
+function VirtualNode(node, shadowId) {
```

Comment:
Any reason not to do something like:

```js
class VirtualNode {
  constuctor(node, shadowId) {
    // ...
  }

  hasClass(className) {
    // ...
  }
}
```

We've [had Babel here since 2016](https://github.com/dequelabs/axe-core/commit/bcf283d72a17f01fb7c9bbfefe0243d69f65a4eb#diff-e56633f72ecc521128b3db6586074d2c), so there shouldn't be any browser compatibility issues.

Reply:
Nope, just following the other base files style. If we're comfortable having our first (and only for now) class, I can do that.